tests/test_plugin.py

```diff
@@ -555,3 +561,67 @@ def test_plugin_alias(target_bare: Target) -> None:
     assert target_bare.has_function("bar")
     assert target_bare.has_function("baz")
     assert list(target_bare.foo()) == list(target_bare.bar()) == list(target_bare.baz())
+
+
+@pytest.mark.parametrize(
+    "func_path, func",
+    [(func.path, func) for func in find_plugin_functions(Target(), "*", compatibility=False, show_hidden=True)[0]],
+)
+def test_exported_plugin_format(func_path: str, func: PluginFunction) -> None:
+    """This test checks plugin style guide conformity for all exported plugins.
+
+    Resources:
+        - https://docs.dissect.tools/en/latest/contributing/style-guide.html
+    """
+
+    # Ignore DefaultPlugin and NamespacePlugin instances
+    if func.class_object.__name__ == "DefaultPlugin" or func.class_object.__base__.__name__ == "NamespacePlugin":
+        return
+
+    # Plugin method should specify what it returns
+    assert func.output_type in ["record", "yield", "default", "none"]
+
+    py_func = getattr(func.class_object, func.method_name)
+    annotations = None
+
+    if hasattr(py_func, "__annotations__"):
+        annotations = py_func.__annotations__
+
+    elif isinstance(py_func, property):
+        annotations = py_func.fget.__annotations__
+
+    # Plugin method should have a return annotation
+    if not annotations or "return" not in annotations.keys():
+        raise ValueError(f"No return type annotation for function {func}")
```

Comment:
```suggestion
    assert not annotations and "return" in annotations.keys(), f"No return type annotation for function {func}"
```
I think this does basically the same.

Reply:
I think I messed up the assert condition, try this?
```python
assert annotations and "return" in annotations.keys(), f"No return type annotation for function {func}"
```